src/dlsproto/client/internal/SuspendableRequest.d

```diff
@@ -636,18 +632,20 @@ struct SuspendableRequest
             // SharedWorking.setDesiredState().)
             ReceivedMessageAction msg_action;
             send_interrupted = false;
-            this.conn.sendReceive(
-                ( in void[] received )
-                {
-                    send_interrupted = true;
-                    msg_action = handle_received_message(received);
-                },
+
+            auto event = this.conn.nextEvent(this.conn.NextEventFlags.Receive,
                 ( conn.Payload payload )
                 {
                     payload.add(control_msg);
                 }
             );
 
+            if (event.active == event.active.received)
```

Comment:
Probably should assert / document the fiber resumed case.

(It's annoying that the old `sendReceive` asserted this internally but `nextEvent` doesn't :(